### PR TITLE
Fix radiation units in converters

### LINKED ISSUE #1365: Hardcoded units in radiation indices
### Generic Issue

Indices `longwave_upwelling_radiation_from_net_downwelling` and  `shortwave_upwelling_radiation_from_net_downwelling` use combinations of `rls, rlds, rss, rsds`. The constraints with the decorators on these variables is "[radiation] = [power]/[area]", as far as I understand, it could something else than "W m-2". But the computation assumes that these units are specifically "W m-2" and sets the units of the output to "W m-2". We should simply not change the units at the end. Also, maybe give a warning if the combined radiation variables have different units.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1365 
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Undoes hard-coding of radiation units

### Does this PR introduce a breaking change?
No.

### CODE REVIEW

File: xclim/indices/_conversion.py [hunk 1]
```diff
@@ -1006,14 +1008,16 @@ def longwave_upwelling_radiation_from_net_downwelling(
...
-    xr.DataArray, [W m-2]
+    xr.DataArray, [same as inputs]
         Surface upwelling thermal radiation (rlus).
     """
-    rls = convert_units_to(rls, rlds)
+    if rls.units != rlds.units:
+        logging.warning(
+            "Radiation units are not synonymous. Units conversion performed."
+        )
+        rls = convert_units_to(rls, rlds)
```
Comment (aulemahal) [inline xclim/indices/_conversion.py:1018]:
```suggestion
    rls = convert_units_to(rls, rlds)
```

We haven't done the warning yet in other indices. I think users _expect_ xclim to convert the differing units. Also, a `convert_units_to` call with identical units should be free!

Comment (Zeitsperre) [reply xclim/indices/_conversion.py:1018]: Valid. Let me revise this PR.

File: xclim/indices/_conversion.py [hunk 2]
```diff
@@ -1000,20 +1000,20 @@ def longwave_upwelling_radiation_from_net_downwelling(
...
-        Surface net thermal radiation [W m-2].
+        Surface net thermal radiation.
     rlds : xr.DataArray
-        Surface downwelling thermal radiation [W m-2].
+        Surface downwelling thermal radiation.
 
     Returns
     -------
-    xr.DataArray, [W m-2]
+    xr.DataArray, [same as inputs]
```
Comment (coxipi) [inline xclim/indices/_conversion.py:1009]: To imitate what `daily_temperature_range` which computes `tasmax - tasmin`:
```suggestion
    xr.DataArray, [same units as rlds]
```

File: xclim/indices/_conversion.py [hunk 3]
```diff
@@ -1026,20 +1026,20 @@ def shortwave_upwelling_radiation_from_net_downwelling(
...
-        Surface net solar radiation [W m-2].
+        Surface net solar radiation.
     rsds : xr.DataArray
-        Surface downwelling solar radiation [W m-2].
+        Surface downwelling solar radiation.
 
     Returns
     -------
-    xr.DataArray, [W m-2]
+    xr.DataArray, [same as inputs]
```
Comment (coxipi) [inline xclim/indices/_conversion.py:1035]:
```suggestion
    xr.DataArray, [same units as rsds]
```

**Review by coxipi — APPROVED** [review]
LGTM!

**Pull request merged**